audio_service_platform_interface/lib/audio_service_platform_interface.dart

```diff
@@ -717,53 +762,52 @@ enum RatingStyleMessage {
 class OnPlaybackStateChangedRequest {
   final PlaybackStateMessage state;
 
-  OnPlaybackStateChangedRequest({
-    required this.state,
-  });
+  @literal
+  const OnPlaybackStateChangedRequest({required this.state});
 
-  factory OnPlaybackStateChangedRequest.fromMap(Map map) =>
+  factory OnPlaybackStateChangedRequest.fromMap(Map<String, dynamic> map) =>
       OnPlaybackStateChangedRequest(
-          state: PlaybackStateMessage.fromMap(map['state']));
+        state:
+            PlaybackStateMessage.fromMap(map['state'] as Map<String, dynamic>),
+      );
 
-  Map<dynamic, dynamic> toMap() => <dynamic, dynamic>{
+  Map<String, dynamic> toMap() => <String, dynamic>{
         'state': state.toMap(),
       };
 }
 
 class OnQueueChangedRequest {
   final List<MediaItemMessage> queue;
 
-  OnQueueChangedRequest({
-    required this.queue,
-  });
+  @literal
+  const OnQueueChangedRequest({required this.queue});
 
-  factory OnQueueChangedRequest.fromMap(Map map) => OnQueueChangedRequest(
+  factory OnQueueChangedRequest.fromMap(Map<String, dynamic> map) => OnQueueChangedRequest(
       queue: map['queue'] == null
           ? []
           : (map['queue'] as List)
-              .map((raw) => MediaItemMessage.fromMap(raw))
+              .map((dynamic raw) => MediaItemMessage.fromMap(_castMap(raw as Map)!))
               .toList());
 
-  Map<dynamic, dynamic> toMap() => <dynamic, dynamic>{
+  Map<String, dynamic> toMap() => <String, dynamic>{
         'queue': queue.map((item) => item.toMap()).toList(),
       };
 }
 
 class OnMediaItemChangedRequest {
   final MediaItemMessage? mediaItem;
 
-  OnMediaItemChangedRequest({
-    required this.mediaItem,
-  });
+  @literal
+  const OnMediaItemChangedRequest({required this.mediaItem});
 
-  factory OnMediaItemChangedRequest.fromMap(Map map) =>
+  factory OnMediaItemChangedRequest.fromMap(Map<String, dynamic> map) =>
       OnMediaItemChangedRequest(
         mediaItem: map['mediaItem'] == null
             ? null
-            : MediaItemMessage.fromMap(map['mediaItem']),
+            : MediaItemMessage.fromMap(_castMap(map['mediaItem'] as Map)!),
```

Comment:
this (and a few others in this file) place might be dangerous, because if someone implements the interface, this (unlikely, but still) might be not `Map`, but let's say `Map<String, dynamic>` if they decide to cast it for some reason

Reply:
I wouldn't consider that dangerous, these casts are effectively like assertions so if they errorred out, they would be doing their job perfectly.

Reply:
(Well, runtime checking is still more dangerous than static checking, but we have no choice until Pigeon is ready to use.)